src/libraries/Native/Unix/System.Native/extra_libs.cmake

```
@@ -13,4 +13,8 @@ macro(append_extra_system_libs NativeLibsExtra)
     if (CLR_CMAKE_TARGET_IOS OR CLR_CMAKE_TARGET_TVOS)
         list(APPEND ${NativeLibsExtra} "-framework Foundation")
     endif ()
+
+    if (CLR_CMAKE_TARGET_LINUX AND HAVE_GETADDRINFO_A)
```

@VSadov in https://github.com/dotnet/runtime/pull/34633#discussion_r545599929 did you like this or just
```suggestion
    if (CLR_CMAKE_TARGET_LINUX)
```
?

If `CLR_CMAKE_TARGET_LINUX` is true, then `HAVE_GETADDRINFO_A` will be true too for libraries-build here?
When that's the case https://github.com/dotnet/runtime/blob/33da8da9dd3c2ceced296876af8bd9189cc3741f/src/libraries/Native/Unix/configure.cmake#L904-L915 could be simplified too.

Now I have it this way, to be on the safe side.

---

Single file host builds in installer partition. It will be moved to clr eventually, but either way it is not in libraries. 
So either `HAVE_GETADDRINFO_A` probe needs to be done in other partitions too, or we could do just `if (CLR_CMAKE_TARGET_LINUX)`

I think `if (CLR_CMAKE_TARGET_LINUX)` would be sufficient. 
Since you are making `anl` a `CMAKE_REQUIRED_LIBRARIES`, we would not even get this far if it is not present.

---

> Single file host builds in installer partition.

That's new information for me -- thanks!